src/utils/generator.js

```diff
@@ -253,10 +253,11 @@ class ModifiedBindingEntry extends GeneratorEntry {
       containingGenerator === this.containingGenerator,
       "This entry requires effects to be applied and may not be moved"
     );
-    invariant(this.modifiedBinding.value === this.newValue);
+    invariant(this.modifiedBinding.value === this.newValue, "ModifiedBinding's value has been improperly modified.");
```

Comment:
This is a good example of an utterly useless comment. Much better (and much harder) is to write a comment that explains to a reader of the code why this invariant is expected to hold. In terms of debug logging, for those who care about such things, it would be better to include information about the two values.